examples/quickstart/matrix_multiplication.cpp

```diff
@@ -0,0 +1,87 @@
+////////////////////////////////////////////////////////////////////////////////
+//  Copyright (c) 2021 Dimitra Karatza
+//
+//  SPDX-License-Identifier: BSL-1.0
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+////////////////////////////////////////////////////////////////////////////////
+
+// Parallel implementation of matrix multiplication
+
+#include <hpx/local/algorithm.hpp>
+#include <hpx/local/future.hpp>
+#include <hpx/local/init.hpp>
+
+#include <iostream>
+#include <random>
+#include <vector>
+
+///////////////////////////////////////////////////////////////////////////////
+//[mul_hpx_main
+int hpx_main()
+{   
+    int rowsA, colsA, rowsB, colsB;
+    rowsA = 2;
+    colsA = 3;
+    rowsB = colsA;
+    colsB = 2;
+    
+    // Initialize matrices A and B
+    std::vector<int> A(rowsA * colsA);
+    std::vector<int> B(rowsB * colsB);
+    std::vector<int> R(rowsA * colsB);
+
+    // Matrices have random values in the range [0, 10]
+    std::random_device rd;
+    std::mt19937 gen(rd());
+    std::uniform_real_distribution<double> dis(0.0, 10.0);
+    hpx::generate(hpx::execution::seq, std::begin(A), std::end(A), std::bind(dis, gen));
+    hpx::generate(hpx::execution::seq, std::begin(B), std::end(B), std::bind(dis, gen));
+
+    // Perform matrix multiplication
+    hpx::for_loop(hpx::execution::par, 0, rowsA, [&](auto i){
+        hpx::for_loop(0, colsB, [&](auto j){
+            R[i * rowsA + j] = 0;
+            hpx::for_loop(0, rowsB, [&](auto k){
+                R[i * rowsA + j] += A[i * rowsA + k] * B[k * rowsA + j];
```

Comment:
The stride for each matrix is currently not consistent. Later when you print the matrices you use `colsX` for some and `rowsX` for some, leading to wrong results. On the above line, the stride of A and B is also not necessarily the same, although they both use `rowsA`. It doesn't matter which one you use, as long as it's consistent (one is row-major, the other column-major).

Reply:
This one still needs fixing (the access to `B` uses `rowsA`). I'd perhaps also define
```
    std::size_t rowsR = rowsA;
    std::size_t colsR = colsB;
```
for indexing into `R`.